rackcity/views/change_plan_views.py

```diff
@@ -138,7 +138,25 @@ def change_plan_remove_asset(request, id):
             status=HTTPStatus.BAD_REQUEST,
         )
     try:
-        asset_cp_model.delete()
+        if asset_cp_object.model.is_blade_chassis():
+            blades = AssetCP.objects.filter(chassis=asset_cp_object, change_plan=change_plan)
+            remove_chassis = True
+            for blade in blades:
+                if len(get_changes_on_asset(blade.related_asset, blade)) == 0:
+                    remove_chassis = False
+            if remove_chassis:
+                asset_cp_object.delete()
+            else:
+                for field in asset_cp.related_asset._meta.fields:
+                    if (
+                            field.name != "id"
+                            and field.name != "assetid_ptr"
+                            and field.name != "chassis"
+                    ):
+                        setattr(asset_cp, field.name, getattr(asset_cp.related_asset, field.name))
+
```

Comment:
if a chassis has blades with changes on the change plan -> dont actually remove it from AssetCP, just make it the same as live .

Reply:
this is super fun